src/tools/immutable_object_cache/ObjectCacheStore.cc

```diff
@@ -0,0 +1,244 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "ObjectCacheStore.h"
+#include "Utils.h"
+#include <experimental/filesystem>
```

Comment:
We are on c++17, so you could use ```<filesystem>```

Reply:
`<filesystem> `seems to be one exception here
- on g++8.x `#include <filesystem>` works fine

- on g++7.x  `i#nclude <filesystem> ` will trigger `fatal error: filesystem: No such file or directory`
- `#include <experimental/filesystem>` works fine on both g++7.x and g++8.x